feat: settings notifications

fbwoolf opened the pull request:
This PR implements screens for settings notifications, adds tabs for 'push' and 'email' notifications, and uses an input to store a preferred email address to receive notifications.

I used zod to validate the email address. I'll need design feedback on what to show for an invalid email address, rn I'm showing a toast but we only have a `success` and `info` types currently. I think I need to add a `fail` type?

<img src="https://github.com/user-attachments/assets/0be42ffb-f348-4fd4-9723-57f21348d20f" width="300" /> <img src="https://github.com/user-attachments/assets/3090a269-5078-4cf0-9939-7549f038aa86" width="300" />
<img src="https://github.com/user-attachments/assets/02d8fe4a-a628-4e9b-8210-0c1178aaec53" width="300" /> <img src="https://github.com/user-attachments/assets/282b3511-ca4b-454e-b2eb-a4e31f5a7cf1" width="300" />
<img src="https://github.com/user-attachments/assets/1273e2c0-f6cb-44b0-a8e9-11721c8ab672" width="300" />

pete-watters commented on apps/mobile/src/app/(home)/settings/notifications/_layout.tsx:
```diff
@@ -0,0 +1,66 @@
+  );
+}
+
+export default function SetttingsNotificationsLayout() {
```
You have a typo here or else you're excited about this feature `Setttings`

pete-watters commented on apps/mobile/src/app/(home)/settings/notifications/email.tsx:
```diff
@@ -0,0 +1,58 @@
+
+  return (
+    <Box flex={1} backgroundColor="ink.background-primary">
+      <ScrollView
```
Not for this PR but we seem to repeat this `<ScrollView` with these props a lot (5 times). We could look into a layout for it?

fbwoolf replied:
I forgot I created a `SettingsScreenLayout` to use here, so will do. Thx for reminder.

pete-watters commented on apps/mobile/src/app/(home)/settings/notifications/_layout.tsx:
```diff
@@ -0,0 +1,66 @@
+import { t } from '@lingui/macro';
+import { Tabs, router, usePathname, useRouter } from 'expo-router';
+
+function HeaderBottomTabs() {
```
Maybe `HeaderBottomTabs` + `NavigationHeader` could be in their own `Header` component?

fbwoolf replied:
Our headers really need some work, so I'll see if there is an issue to coordinate all these a bit better. If not, I'll make one.

pete-watters replied:
Thanks @fbwoolf 👍 It seems like we have the headers we need but they could do with some refactoring

pete-watters reviewed (approved):
Great work 🚀  

I added suggestions but it's fine as is, bar typo

kyranjamie commented on apps/mobile/src/store/settings/settings.ts:
```diff
@@ -54,8 +57,8 @@ export const initialState: SettingsState = {
   bitcoinUnitPreference: 'bitcoin',
+  emailAddressPreference: '',
   fiatCurrencyPreference: 'USD',
-  createdOn: new Date().toISOString(),
```
Why is this removed? This is useful metadata to keep even if we don't display anywhere.

fbwoolf replied:
I searched and it wasn't being used, so I thought it was just done temporarily for dev purposes. I can add it back. I think at one point it was being used as a placeholder where designers want us to show the app version number? We have the same prop on wallet state.

fbwoolf commented on apps/mobile/src/app/(home)/settings/notifications/email.tsx:
```diff
@@ -0,0 +1,58 @@
+          gap: theme.spacing[5],
+        }}
+      >
+        <TouchableOpacity
+          onPress={() => {
+            emailAddressSheetRef.current?.present();
+          }}
+        >
+          <Flag
+            img={
+              <Avatar>
+                <EmailIcon />
+              </Avatar>
+            }
+          >
+            <ItemLayout
+              actionIcon={<ChevronRightIcon variant="small" />}
+              captionLeft={t`Awaiting verification`}
+              titleLeft={t`Email address`}
+            />
+          </Flag>
+        </TouchableOpacity>
```
The opacity does drop on my other uses of this, so let me check why different here.

fbwoolf replied:
I tested this and the opacity change DOES work for me, so not sure why wasn't for you.

fbwoolf commented:
Updated the toast types here cc @fabric-8, also the copy on the success was to wordy so I simplified to fit on screen better...

<img src="https://github.com/user-attachments/assets/7b83555a-e1e2-40b0-afa9-7318a9ab0ba5" width="300" /> <img src="https://github.com/user-attachments/assets/9ae88eff-a428-4de7-96a4-e009a49695f0" width="300" />
